Investigate build-core-pypy3 failures

Builds are intermittently failing the `build-core-pypy3` job. This looks like a timing issue in the getting started tests run under pypy3:

```
>           raise MaxRetryError(_pool, url, error or ResponseError(cause))
E           urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=5000): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x00007f0c072ad2b8>: Failed to establish a new connection: [Errno 111] Connection refused',))

../../../.tox/pypy3-test-core-getting-started/site-packages/urllib3/util/retry.py:439: MaxRetryError
```
Full log:

https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-python/733/workflows/ed2f57e1-11c5-4b92-9f3a-06ece287e97d/jobs/15242

--- a/docs/getting_started/tests/test_flask.py
+++ b/docs/getting_started/tests/test_flask.py
@@ -29,10 +29,7 @@ class TestFlask(unittest.TestCase):
         server = subprocess.Popen(
             [sys.executable, server_script], stdout=subprocess.PIPE,
         )
-        retry_strategy = Retry(
-            total=10,
-            backoff_factor=1,
-        )
+        retry_strategy = Retry(total=10, backoff_factor=1)
         adapter = HTTPAdapter(max_retries=retry_strategy)
         http = requests.Session()
         http.mount("http://", adapter)

--- a/docs/getting_started/tests/test_flask.py
+++ b/docs/getting_started/tests/test_flask.py
@@ -18,6 +18,8 @@ import unittest
 from time import sleep
 
 import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
 
 
 class TestFlask(unittest.TestCase):
@@ -27,10 +29,16 @@ class TestFlask(unittest.TestCase):
         server = subprocess.Popen(
             [sys.executable, server_script], stdout=subprocess.PIPE,
         )
-        sleep(1)
+        retry_strategy = Retry(
+            total=10,
+            backoff_factor=1,
+        )
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        http = requests.Session()
+        http.mount("http://", adapter)
 
         try:
-            result = requests.get("http://localhost:5000")
+            result = http.get("http://localhost:5000")
             self.assertEqual(result.status_code, 200)
 
             sleep(0.1)

